subprojects/ivy/src/main/java/org/gradle/api/publish/ivy/internal/publisher/IvyDescriptorFileGenerator.java

```diff
@@ -64,76 +66,111 @@ public void execute(XmlProvider xmlProvider) {
         }
     };
 
```

Comment:
Here, I extracted a "Model" to encapsulate all the data we require to generate an Ivy descriptor, and a ModelWriter separately from the generator. One benefit of that is that the design could look a lot like the design for MavenPomFileGenerator, which relied on Maven's own Model/ModelWriter code.